packages/shared/src/utils/runtimeEnvironment.ts

```diff
@@ -1,30 +1,83 @@
-export const isDevelopmentEnvironment = (): boolean => {
-  try {
-    return process.env.NODE_ENV === 'development';
-    // eslint-disable-next-line no-empty
-  } catch (err) {}
+import { parsePublishableKey } from './keys';
 
-  // TODO: add support for import.meta.env.DEV that is being used by vite
+/**
+ * Check for development runtime environment using the environment variables of the Clerk publishableKey.
+ *
+ * Examples:
+ *
+ * 1. using process.env: `isDevelopmentEnvironment()`
+ * 2. passing custom process.env source: `isDevelopmentEnvironment(import.meta.env)`
+ * 3. using publishableKey: `isDevelopmentEnvironment('pk_********')`
+ *
+ * @param envOrPublishableKey
+ * @returns boolean
+ */
+export const isDevelopmentEnvironment = (envOrPublishableKey?: EnvironmentOrPublishableKey): boolean => {
+  return isEnvironment('development', envOrPublishableKey);
+};
 
-  return false;
+/**
+ * Check for testing runtime environment using the environment variables of the Clerk publishableKey.
+ *
+ * Examples:
+ *
+ * 1. using process.env: `isTestEnvironment()`
+ * 2. passing custom process.env source: `isTestEnvironment(import.meta.env)`
+ * 3. using publishableKey: `isTestEnvironment('pk_test_********')`
+ *
+ * The `isTestEnvironment()` will always be false for any publishableKey since development and production
+ * publishable keys are only supported.
+ *
+ * @param envOrPublishableKey
+ * @returns boolean
+ */
+export const isTestEnvironment = (envOrPublishableKey?: EnvironmentOrPublishableKey): boolean => {
+  return isEnvironment('test', envOrPublishableKey);
 };
 
-export const isTestEnvironment = (): boolean => {
-  try {
-    return process.env.NODE_ENV === 'test';
-    // eslint-disable-next-line no-empty
-  } catch (err) {}
+/**
+ * Check for production runtime environment using the environment variables of the Clerk publishableKey.
+ *
+ * Examples:
+ *
+ * 1. using process.env: `isProductionEnvironment()`
+ * 2. passing custom process.env source: `isProductionEnvironment(import.meta.env)`
+ * 3. using publishableKey: `isProductionEnvironment('pk_live_********')`
+ *
+ * @param envOrPublishableKey
+ * @returns boolean
+ */
+export const isProductionEnvironment = (envOrPublishableKey?: EnvironmentOrPublishableKey): boolean => {
+  return isEnvironment('production', envOrPublishableKey);
+};
 
-  // TODO: add support for import.meta.env.DEV that is being used by vite
-  return false;
+const getRuntimeFromProcess = (env?: typeof process.env) => {
+  return env?.NODE_ENV;
```

Comment:
❓ @dimkl this one applies only on node environments, right?
Or, if you pass an `env` object, it should include a `NODE_ENV` property for this to work, right?

Reply:
No, it applies for all runtimes (Node, Cloudflare Workers, ...).
If you pass an `env` object with the `NODE_ENV` property, it should work.
I probably need to test if the `process.env` type works for Vite and Cloudflare workers.

Reply:
@dimkl That's what I meant. You need to have `NODE_ENV` in your passed object for this to work. `NODE_ENV` usually refers to node environments.

Reply:
I agree that this name is a bit misleading. At the moment the function assumes that `NODE_ENV` exists and that it's Node.js

You can do a check for the runtime like this: https://github.com/unjs/std-env/blob/main/src/runtimes.ts

Probably the whole "runtime" name (also for the file name) is a bit misleading as we only check the `NODE_ENV` environment 🤔